services/spar/test-integration/Test/MetricsSpec.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TupleSections       #-}
+{-# LANGUAGE ViewPatterns        #-}
+
+-- | See also: services/brig/test/integration/API/Metrics.hs
+module Test.MetricsSpec (spec) where
+
+import Imports
+import Bilge
+import Control.Lens
+import Data.String.Conversions (cs)
+import Data.Aeson
+import Data.Aeson.Lens
+import Data.Set as Set
+import Util
+
+
+-- | path parsing works; http verb, captured variables are not distinguished in the metrics.
+spec :: HasCallStack => SpecWith TestEnv
+spec = describe "metrics" . it "works" $ do
+  spar <- asks (^. teSpar)
+  let p1 = "sso/metadata"
+      p2 idpid = "sso/initiate-login/" <> idpid
+
+  _ <- call $ get (spar . path p1)
+  _ <- call $ get (spar . path (p2 "316f1c18-2980-11e9-ab0b-ef604d1791b2"))
+  _ <- call $ get (spar . path (p2 "60a7dda8-2980-11e9-b359-fb5b41565453"))
+
+  resp :: Value <- call $ jsonBody <$> get (spar . path "i/monitoring")
```

Comment:
> the exact same end-point should be provided by the exact same middleware in both brig and spar

Based on false assumptions.

This endpoint is not provided by the middleware, but defined manually. See [here in brig](https://github.com/wireapp/wire-server/blob/develop/services/brig/src/Brig/API.hs#L119-L120) for instance.

There are two relevant endpoints for metrics:

1) `/i/monitoring` - this gives JSON in a collectd style format. We want to deprecate this, and use the prometheus format instead. That's why it was not introduced to spar. This endpoint exists in all services **except spar**.
2) `/metrics` - this gives prometheus-style new-line delimited TEXT (not json). This endpoint (so far, before this PR) exists only in spar, and in no other service. This test `MetricsSpec` is incorrect as it tests for output of an undesired endpoint (for spar) in an undesired format.

Reply:
thanks, i was indeed confused about this context.